src/DdSdkReactNative.tsx

```diff
@@ -37,21 +37,23 @@ export class DdSdkReactNative {
 
             DdSdk.initialize(
                 new DdSdkConfiguration(
-                    configuration.clientToken, 
-                    configuration.env, 
+                    configuration.clientToken,
+                    configuration.env,
                     configuration.applicationId,
                     configuration.nativeCrashReportEnabled,
                     configuration.sampleRate,
                     configuration.site,
                     configuration.trackingConsent,
                     configuration.additionalConfig
                 )
-            )
-            DdSdkReactNative.enableFeatures(configuration)
-            DdSdkReactNative.wasInitialized = true
-            resolve()
-        }))
-
+            ).then(() => {
+                DdSdkReactNative.enableFeatures(configuration)
+                DdSdkReactNative.wasInitialized = true
+                resolve()
+            }, (rejection) => {
+                reject(rejection)
+            })
+        })
```

Comment:
you can see how react-native finds native modules and initializes them in [this commit](https://github.com/facebook/react-native/commit/ff66600224e78fec5d0e902f8a035b78ed31a961)

in short, native modules are initialized in the order that iOS runtime loads them and we can't change this order.

so this change doesn't fix the problem but doesn't do any harm neither, we can still merge 👌

Reply:
> you can see how react-native finds native modules and initializes them in [this commit](https://github.com/facebook/react-native/commit/ff66600224e78fec5d0e902f8a035b78ed31a961)

I'm wondering if this may change to AOT module loading with, say turbo modules, or at any point in future. So in this case related PR https://github.com/DataDog/dd-bridge-ios/pull/17 still makes sense.